compaction/task_manager_module.cc

```diff
@@ -189,7 +189,7 @@ struct table_tasks_info {
     {}
 };
 
-future<> run_on_table(sstring op, replica::database& db, std::string keyspace, table_info ti, std::function<future<> (replica::table&)> func) {
+future<> run_on_table(std::string_view op, replica::database& db, std::string keyspace, table_info ti, std::function<future<> (replica::table&)> func) {
```

Comment:
With this change it's no longer possible to do 

```c++
    sstring op = ...;
    (void)run_on_table(op, ...);
```

No code does it now, but it's a time bomb

Reply:
if it explodes without being awared, then it's a bomb. but IMHO, this is not. also, one can always get an `std::string_view` from `seastar::sstring` for free, right?

Reply:
> one can always get an std::string_view from seastar::sstring for free, right?

Right, but what if original sstring is freed, does string_view continue holding its buffer?

Reply:
Yes it's better to copy here.

Reply:
oh, i see what you mean now. you were afraid that we can pass an `sstring` without noticing that it is returning a shallow copy, and the `sstring` is not kept alive by the caller. well, that's true.

i don't feel strong either way. as this is not a critical path. i just don't want to pay for something i don't use.

but if you believe we should **always** pass a string by value, i will close this PR. but still, we always have to check the callee's function signature before using it.